assets/js/homepage.js

```diff
@@ -4,43 +4,29 @@
  * Handles behaviour of the homepage featured image
  */
 ( function() {
-
-	var homepageContent = '.page-template-template-homepage .type-page.has-post-thumbnail';
-
-	if ( ! jQuery( homepageContent ).length ) {
-		// Only apply layout to the homepage content component if it exists on the page
-		return;
-	}
+	var homepageContent = null;
+	var siteMain = null;
 
 	/**
-	 * Set the hero component dimensions and positioning
+	 * Set hero content dimensions / layout
+	 * Run adaptive backgrounds and set colors
 	 */
-	function homepageContentDimensions() {
-		var windowWidth	    = jQuery( window ).width();
-		var offset          = jQuery( '.site-main' ).offset();
+	requestAnimationFrame(function() {
```

Comment:
Previously, the implementation would wait for the document to be ready. By using `requestAnimationFrame`, we  can assume that the first frame is about to be rendered, aka the parsing of the document is done.
This shouldn't be necessary, as the homepage script is injected at the end of the document (hence the DOM is already created and can be queried).